x/evm/keeper/abci.go

```diff
@@ -26,6 +28,13 @@ import (
 // BeginBlock sets the sdk Context and EIP155 chain id to the Keeper.
 func (k *Keeper) BeginBlock(ctx sdk.Context, _ abci.RequestBeginBlock) {
 	k.WithChainID(ctx)
+
+	if utils.IsEthermintDevChain(ctx) {
+		// trigger VFBC registration on Ethermint devnet for development purpose
+		if err := k.DeployVirtualFrontierBankContractForAllBankDenomMetadataRecords(ctx, nil); err != nil {
```

Comment:
IMO it's better not to panic on beginBlock unless it's store corruption.
especially if iterating over multiple object and one of them can fail due to app logic

Reply:
This will leading to dirty state, as state may changed during execution. Ignoring it (by logging only and keep going) is dangerous.

Reply:
@mtsitrin [here](https://github.com/dymensionxyz/ethermint/pull/11/commits/da3a1486e31914d13cf0dbc195c342c0dd4dd66d)